example/ndpiReader.c

```diff
@@ -92,14 +92,21 @@ static char* domain_to_check = NULL;
 static char* ip_port_to_check = NULL;
 static u_int8_t ignore_vlanid = 0;
 /** User preferences **/
-u_int8_t enable_realtime_output = 0, enable_protocol_guess = 1, enable_payload_analyzer = 0, num_bin_clusters = 0, extcap_exit = 0;
+u_int8_t enable_realtime_output = 0, enable_protocol_guess = NDPI_GIVEUP_GUESS_BY_PORT | NDPI_GIVEUP_GUESS_BY_IP, enable_payload_analyzer = 0, num_bin_clusters = 0, extcap_exit = 0;
 u_int8_t verbose = 0, enable_flow_stats = 0;
-int nDPI_LogLevel = 0;
-char *_debug_protocols = NULL;
+
+struct cfg {
+  char *proto;
+  char *param;
+  char *value;
+};
+#define MAX_NUM_CFGS 32
+static struct cfg cfgs[MAX_NUM_CFGS];
```

Comment:
Is it worth to make this a dynamic allocated array depending on the amount of used config params?

Reply:
I think we can safely ignore that: 32 is plenty of space, the limit is always checked and we are in the application example... [if that code had been in the library, I would have used the realloc stuff...]